apis/externalsecrets/v1beta1/secretstore_vault_types.go

```diff
@@ -80,6 +88,18 @@ type VaultProvider struct {
 	ForwardInconsistent bool `json:"forwardInconsistent,omitempty"`
 }
 
+// VaultClientTLS is the configuration used for client side related TLS communication,
+// when the Vault server requires mutual authentication.
+type VaultClientTLS struct {
+	// ClientCert is a certificate added to the transport layer
+	// when communicating with the Vault server.
+	ClientCert *esmeta.SecretKeySelector `json:"clientCert,omitempty"`
+
+	// SecretRef to a key in a Secret resource containing client private key
+	// added to the transport layer when communicating with the Vault server.
+	SecretRef *esmeta.SecretKeySelector `json:"secretRef,omitempty"`
+}
```

Comment:
While i like having the flexibility with having separate key/cert fields i would prefer to lookup using well-known keys `tls.key` and `tls.crt` to match the `secret.type=tls` scheme.

This may seem a bit magical, however i think that 80% of users have the tls cert/key stored in the same secret.

I don't mind keeping the approach, this is just a opinion. If you'd prefer to keep it i suggest to name it more clearly, maybe `certSecretRef` and `keySecretRef`?

What do you think?

Reply:
HI @moolen , thanks for your suggestion, I understand your proposal about using the well-known keys and focusing on convention over configuration, it does make sense to me, however, since the the "TLS certificates authentication" section under the `auth.cert` attribute is already allowing defining the `key`, I believe not allowing it for the `mtls` part would create an inconsistent API. In this case, what we can actuality do, is to make the `key` attribute optional and if the user does not provide it, the well-known key values will be used.

What do you think? If agreed, I will add a short note about the behaviour to the comments/documentation as well.

Reply:
@moolen , I applied a mix from your suggestion and mine from above, the attributes were renamed to `certSecretRef` and `keySecretRef` as you suggested, and the keys are now optional, if the user does not provided it, the well-known keys `tls.key` and `tls.crt` will be used. What do you think?

Reply:
> I believe not allowing it for the mtls part would create an inconsistent API

Fully agree!

Also your proposal sounds good to me! :+1: